Collected bugfixes and updates for the next patch release

akohlmey opened the pull request:
## Purpose

This pull request combines accumulated small bugfixes and updates

## Author(s)

Axel Kohlmeyer (Temple U)
Emile Maras (CEA, France)
Dr. Alban de Vaucorbeil, Monash University, Clayton, VIC, Australia
Stefan Paquay, (Brandeis U)

## Backward Compatibility

N/A

## Implementation Notes

- address a compilation issue for fix `neigh/history/omp` with `-DLAMMPS_BIGBIG`
- ditto for pair style `airebo/intel`, however, there are still fundamental issues leading to "data slicing" in `USER-INTEL`, so that compilation will now error out in compiling `fix_intel.cpp` with `-DLAMMPS_BIGBIG` defined; similar to `fix atc` from `USER-ATC`.
- bugfix/update for NEB from Emile Maras (see git log for details)
- bugfixes for `write_data` output with atom style `smd`. taken from PR #745 plus additional changes. this closes #745.
- documentation bugfix for `fix property/atom`
- add call to `neighbor->init()` at the end of methods in `delete_atoms` and `create_bonds` that request a neighbor list. thus will clear all neighbor list requests and thus avoid having the per-comment neighbor list being considered for re-use in the next run command.
- consistently refer to `filelink.o` in LATTE related docs and README files.
- avoid multiple cases of division by zero when using Ewald or PPPM on an empty or uncharged system. require using `kspace_modify gewald` in those cases as the estimator will fail.
- correctly tally added bonds/angles/dihedrals into global counters with `create_bonds single/*`, so that data files have correct headers. fixes bug reported by bernard rousseau on lammps-users: https://sourceforge.net/p/lammps/mailman/message/36180184/

## Post Submission Checklist

_Please check the fields below as they are completed_
- [x] The feature or features in this pull request is complete
- [x] Suitable new documentation files and/or updates to the existing docs are included
- [x] The source code follows the LAMMPS formatting guidelines

akohlmey commented:
@v0i0 and @wmbrownIntel  please have a look at the changes to `airebo/intel` here. they will make the code compile when tagint is set to a 64-bit integer through `-DLAMMPS_BIGBIG`, but i doubt they are sufficient to run this correctly. probably not even for small systems. for example, there is the use of the `ivec` data type for tagint data, which i didn't know how to correct. nevertheless, rather than having a compile time error, that is difficult for some users to handle, i'd prefer to either have this sorted out or then add a run-time error in the airebo/intel constructor to stop when `sizeof(tagint) != sizeof(int)`.

v0i0 commented:
@akohlmey The changes in airebo/intel look fine (even better: remove the unused variables, i.e. 2168, 2170, 3643, 4493, 4524), and I would not expect any problems from the changes. The code should still be correct.

However, there is a real issue with `-DLAMMPS_BIGBIG` and USER-INTEL in general.
In the neighbor list building routines in `npair_full_bin_ghost_intel.cpp` and `npair_intel.cpp`, `tagint * tag` data is pushed through a `int * ttag` buffer (search for `get_ncachetag`).

akohlmey commented:
@v0i0 thanks for taking a look. i had similar suspicions. so the prudent route is to completely disable use of -DLAMMPS_BIGBIG. i've implemented it - for now - as a compile time error, but - if needed - this could be changed to a run-time error with all the changes in place. i've also removed the dead code. there is likely more, but tracking those down is tedious with all the macros and templates.
